soc: NXP:  RT101x: flash choosen node make all RT101x seriese out of flash

**Describe the bug**
all NXP mimxrt101x series board build failure due to the commit 48214e86b08116d9029eca598ba521d67d2772f2

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
   mimxrt1015_evk/mimxrt1010_evk
 - What have you tried to diagnose or workaround this issue?
   revert the #48214e86b08116d9029eca598ba521d67d2772f2 can make build pass
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
   yes   #48214e86b08116d9029eca598ba521d67d2772f2
 
**To Reproduce**
Steps to reproduce the behavior:

```
$ cd samples/hello_world

$ west build -b mimxrt1015_evk
```

**Expected behavior**
build and run pass

**Impact**
all RT101x series platform

**Logs and console output**
```
[2/153] Generating include/generated/version.h
-- Zephyr version: 3.3.99 (/home/shared/disk/zephyr_project/zephyr_test/zephyr), build: zephyr-v3.3.0-1139-g1bc6045fd9e5
[143/153] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/hello_world/build/zephyr/zephyr_pre0.map
: && ccache /home/ubuntu/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/hello_world/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/boards/arm/mimxrt1015_evk/libboards__arm__mimxrt1015_evk.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_nxp/hal_nxp/lib..__modules__hal__nxp.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/home/ubuntu/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/thumb/v7e-m/nofp"  -L/home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/hello_world/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -mcpu=cortex-m7  -mthumb  -mabi=aapcs  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/hello_world/build/zephyr && /home/ubuntu/.local/lib/python3.10/site-packages/cmake/data/bin/cmake -E echo
/home/ubuntu/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre0.elf section `rom_start' will not fit in region `FLASH'
/home/ubuntu/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: region `FLASH' overflowed by 29336 bytes
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/ubuntu/.local/bin/cmake --build /home/shared/disk/zephyr_project/zephyr_test/zephyr/samples/hello_world/build
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, )
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: zephyr-v3.3.0-1420-g9889e8e20a4f

--- a/boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts
+++ b/boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts
@@ -23,6 +23,7 @@
 		zephyr,itcm = &itcm;
 		zephyr,console = &lpuart1;
 		zephyr,shell-uart = &lpuart1;
+		zephyr,flash = &at25sf128a;
 	};
 
 	leds {

--- a/boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts
+++ b/boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts
@@ -23,6 +23,7 @@
 		zephyr,itcm = &itcm;
 		zephyr,console = &lpuart1;
 		zephyr,shell-uart = &lpuart1;
+		zephyr,flash = &at25sf128a;
 	};
 
 	leds {

